default.nix

```diff
@@ -1,27 +1,48 @@
+let
+  inNixShell = builtins.getEnv "IN_NIX_SHELL" != "";
```

Comment:
I weakly prefer to do this via having a separate shell.nix file, which just imports and invokes this default.nix file with an overridden `includeOcamlformat` argument or something